SPEC.md

```diff
@@ -0,0 +1,28 @@
+# Navigation.js specification
+
+## Purpose
+The goal of this specification is to outline the rules and ideas this library adheres to guide a user along a route.
+
+### Units
+By default, the metric system should be used where available.
+
+### Concepts
+Navigation.js looks at a single step along a greater route. All calculations are done against this step and not against the whole route. Once a step is completed, the next step on the route can be compared. If the user is no longer on the current step, the developer should re-route the user and also set their internal step counter back to 0.
+
+### Re-routing
+If at any time the user is not within `45 meters` of any point on the current step, re-route the user.
+
+### Completing a step
+* If the user is within 15 meters of the end of the step, the user has completed the current step.
```

Comment:
What if the step is 16 meters long? Declaring the step complete after one meter would be unhelpful.

Instead, I think we should consider the step to be complete when the user _leaves_ the maneuver zone, not when they enter it. While the user is within the 15 meters (which should also depend on the mode of transportation), we should deliver the highest alert possible. The use case is that the user is at the intersection; they need to be reminded to turn _now_ and not be distracted with details about the following step.